### PR TITLE
Created Class to Check AbsoluteEncoder Readings

### PR DESCRIPTION
**Pull request by avakimc**
* checkLastFive() compares the last five readings of an absolute encoder and returns true if they're all equal
* checkLastFive() is called for each absolute encoder in each subsystem in periodic()

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 1]
```diff
@@ -146,12 +148,7 @@ public void initialize() {
   }
 
   public void rezeroIntake() {
-    deployMotorEncoder.setPosition(
-        AngleUtil.wrapAngle(
-            deployMotorAbsoluteEncoder.getPosition()
-                - Cal.Intake.ABSOLUTE_ENCODER_START_POS_DEG
-                + Cal.Intake.STARTING_POSITION_DEGREES));
-
+    deployMotorEncoder.setPosition(deployMotorAbsoluteEncoderChecker.getMedian());
```
Comment (balghane) [inline src/main/java/frc/robot/subsystems/Intake.java]: So `deployMotorAbsoluteEncoderChecker.getMedian()` really only replaces `deployMotorAbsoluteEncoder.getPosition()`, we still need the other parts

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 2]
```diff
@@ -339,18 +343,10 @@ public void initialize() {
...
 
     // Set elevator encoder position from absolute encoders
-    double elevatorDutyCycleEncodersDifferenceDegrees =
-        AngleUtil.wrapAngleAroundZero(
-            (elevatorLeftAbsEncoder.getPosition() - elevatorRightAbsEncoder.getPosition()));
-    elevatorLeftEncoder.setPosition(
-        (elevatorDutyCycleEncodersDifferenceDegrees
-                * Constants.Lift.ELEVATOR_MOTOR_ENCODER_DIFFERENCES_SCALAR_INCHES_PER_DEGREE)
-            - Cal.Lift.ELEVATOR_ABS_ENCODER_POS_AT_START_INCHES);
+    elevatorLeftEncoder.setPosition(elevatorLeftAbsEncoderChecker.getMedian());
```
Comment (balghane) [inline src/main/java/frc/robot/subsystems/Lift.java]: Same comment as for intake, we still need all the math, just replacing the `Encoder.getPosition()` bits with the filtered value

File: src/main/java/frc/robot/utils/AbsoluteEncoderChecker.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    /** @return true if all of the last five absolute encoder readings were equal, false if otherwise*/
+    public boolean checkLastFive() {
+        for (int i = 1; i < lastFive.length; i ++) {
+            if (lastFive[i] != lastFive[0]) {
+                return false;
+            }
+        }
+        return true;
+    }
```
Comment (balghane) [inline src/main/java/frc/robot/utils/AbsoluteEncoderChecker.java]: Maybe change to "encoder connected" and return `true` if any of them are different (flipped from current logic)? Then the sendable text can be like "Deploy encoder connected"

File: src/main/java/frc/robot/utils/AbsoluteEncoderChecker.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+    /** @return the median of the five current values */
+    public double getMedian() {
+        // resets so only the most up-to-date values are checked
+        medianFilter.reset();
+
+        // adds each of the first four readings to the medianFilter window
+        for (int i = 0; i < lastFive.length-1; i ++) {
+            medianFilter.calculate(lastFive[i]);
```
Comment (balghane) [inline src/main/java/frc/robot/utils/AbsoluteEncoderChecker.java]: Can you just do `medianFilter.calculate()` in addReading and save the latest result value to return in this function?

Comment (avakimc) [reply src/main/java/frc/robot/utils/AbsoluteEncoderChecker.java]: Would we ever need to reset the medianFilter's window? If so, how should I check for that situation?

File: src/main/java/frc/robot/utils/AbsoluteEncoderChecker.java (hunk 3 above)
Comment (avakimc) [reply src/main/java/frc/robot/utils/AbsoluteEncoderChecker.java:33]: Like change the comment to "return true if encoder connected" or change the name of the function to encoderConnected?

File: src/main/java/frc/robot/utils/AbsoluteEncoderChecker.java (hunk 4 above)
Comment (balghane) [reply src/main/java/frc/robot/utils/AbsoluteEncoderChecker.java]: I don't think we ever need to reset, we can just keep feeding in measurements and it'll drop the ones we don't need on its own

**Review by balghane — APPROVED** [review]

**Pull request merged**